src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons.scalatest
+
+import java.{util => ju}
+
+import org.scalatest.{BeforeAndAfter, Suite}
+import za.co.absa.commons.reflect.ReflectionUtils
+
+import scala.collection.JavaConverters._
+
+trait EnvFixture extends BeforeAndAfter {
+  this: Suite =>
+
+  private val m = ReflectionUtils.extractFieldValue[ju.Map[String, String]](System.getenv, "m")
```

Comment:
It seems to me, that this map `m` is what is expected to be found via `System.getenv`, but there is no `System.setX` involved, the content of `m` never actually gets to the environment properties.

Therefore, at least currently, the test suite does not work
https://github.com/AbsaOSS/commons/pull/45/files#diff-be115859ff93a482acdc11a331867bc9c0eadf80fde0762bf2907acd0010b15aR28-R33

Reply:
It works on the CI and my local. What java version are you testing it on? 
The `m` is an object reference to the underlying map that holds all the environment variables. It isn't supposed to be modified from Java API, that's why I used reflection.

Reply:
> so the content of m never actually gets to the environment properties

It's not meant to be propagated to the real system environment, of cause. The goal is to fool `System.getenv()` for testing purposes.

Reply:
I am using
```OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_242-b08)```

It really fails for me, that why I assumed that you just missed something there:
```
EnvFixtureSpec:
- should set environment variable *** FAILED ***
  null did not equal "42" (EnvFixtureSpec.scala:31)
```

Reply:
That's even more interesting.
```
openjdk version "1.8.0_275"
OpenJDK Runtime Environment (build 1.8.0_275-b01)
OpenJDK 64-Bit Server VM (build 25.275-b01, mixed mode)
```
![image](https://user-images.githubusercontent.com/795479/101164024-cf5b5280-3634-11eb-9895-8dbbe3c87348.png)

Reply:
Ah, you are using `AdoptOpenJDK`. It probably makes difference

Reply:
@dk1844 I have tested it in 8 different JDKs and the test is passed everywhere:
- Open JKD - 1.8, 11
- Adopt (both hotspot and openj9) - 1.8, 11
- Amazon corretto - 1.8, 11

All were the latest public patch versions, including (AdoptOpenJDK)(build 1.8.0_275-b01)

Reply:
Works for me, both in IDE and `mvn test`
```
openjdk version "1.8.0_222"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_222-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.222-b10, mixed mode)
```

Reply:
After updating JDK, still works for me
```
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)
```

Reply:
On Windows different collection is used for `getenv(String)` and `getenv()`

```
    static String getenv(String name) {
        return theCaseInsensitiveEnvironment.get(name);
    }

    static Map<String,String> getenv() {
        return theUnmodifiableEnvironment;
    }
```

Reply:
Case insensitivity :facepalm:

Reply:
Nice find, @cerveada!

Having updated to `OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)`, of course there is no change in behavior.